client/lib/plans/features-list.js

```diff
@@ -348,7 +348,7 @@ export const FEATURES_LIST = {
 
 	[ constants.FEATURE_JETPACK_ESSENTIAL ]: {
 		getSlug: () => constants.FEATURE_JETPACK_ESSENTIAL,
-		getTitle: () => i18n.translate( 'Jetpack Essential Features' ),
+		getTitle: () => i18n.translate( 'Jetpack essential features' ),
```

Comment:
:warning: This change will be queued for retranslation. We'll display the English text until then.